### PR TITLE
feat: support indexing in mixed discrete-continuous systems

### PR DESCRIPTION
**Pull request by AayushSabharwal**
## Checklist

- [ ] Appropriate tests were added
- [ ] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [ ] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [ ] Any new documentation only uses public API

### CODE REVIEW

File: src/parameter_indexing.jl [hunk 1]
```diff
@@ -15,6 +15,56 @@ parameter_values(arr::AbstractArray) = arr
 parameter_values(arr::AbstractArray, i) = arr[i]
 parameter_values(prob, i) = parameter_values(parameter_values(prob), i)
 
+"""
+    parameter_values_at_time(p, t)
+
+Return an indexable collection containing the value of all parameters in `p` at time index
+`t`. This is useful when parameter values change during the simulation
+(such as through callbacks) and their values are saved. `t` is the time index in the
```
Comment (baggepinnen) [inline src/parameter_indexing.jl:23]: > `t` is the time index

This sounds ambiguous, is `t` time or index?

If it's an index, perhaps we could clarify by calling it `ti` and say that it's an integer?

File: src/parameter_indexing.jl [hunk 2]
```diff
@@ -15,6 +15,56 @@ parameter_values(arr::AbstractArray) = arr
...
+implemented for [`getu`](@ref) to work correctly.
+"""
+function parameter_values_at_time end
+parameter_values_at_time(p, t) = parameter_values(p)
+
+"""
+    parameter_values_at_state_time(p, t)
+
+Return an indexable collection containing the value of all parameters in `p` at time
+index `t`. This is useful when parameter values change during the simulation (such as
```
Comment (baggepinnen) [inline src/parameter_indexing.jl:41]: same here

File: src/parameter_indexing.jl (hunk 1 above)
Comment (AayushSabharwal) [reply src/parameter_indexing.jl:23]: I'll refactor it to be `i`

Comment (AayushSabharwal) [reply src/parameter_indexing.jl]: Fixed in https://github.com/SciML/SymbolicIndexingInterface.jl/pull/53/commits/cc5da9083caa15e894a4ea2f8d35300b0fb27ff4

File: src/parameter_indexing.jl (hunk 2 above)
Comment (AayushSabharwal) [reply src/parameter_indexing.jl]: Fixed in https://github.com/SciML/SymbolicIndexingInterface.jl/pull/53/commits/cc5da9083caa15e894a4ea2f8d35300b0fb27ff4

**Comment by ChrisRackauckas** [comment]
Docs build fails https://github.com/SciML/SymbolicIndexingInterface.jl/actions/runs/8187637707/job/22388543154?pr=53#step:5:47

**Pull request merged**